lib/p2p/Peer.ts

```diff
@@ -262,10 +272,9 @@ class Peer extends EventEmitter {
     const packet = this.sendHello();
 
     if (!this.handshakeState) {
-      // using the packet type as the response key,
-      // to allow waiting for any incoming HelloPacket
-      await this.wait(packet.type, Peer.RESPONSE_TIMEOUT);
-      assert(this.handshakeState);
+      // wait for an incoming HelloPacket
+      await this.wait(PacketType.HELLO, Peer.RESPONSE_TIMEOUT);
+      assert(isHandshakeState(this.handshakeState));
```

Comment:
I figured you added `isHandshakeState` to validate the runtime object, but with this line I can crash other xud's by sending them `Hello` packet with a partial body. `assert` should be used to enforce integrity on scenarios that should never happen (and if they do, xud should crash).

Reply:
Good point, although technically I think a peer could crash xud by sending an empty packet body even with the previous `assert(this.handshakeState)` line, right? My thinking was this line was to verify that we received the fields we expected. I can definitely change it back, but we will probably want some explicit checking on the body of received packets, discarding malformed packets and penalizing the peer. That can be a separate PR though. I can create an issue if it makes sense to you.